pkg/gofr/service/new_test.go

```diff
@@ -74,7 +74,7 @@ func TestHTTPService_createAndSendRequest(t *testing.T) {
 	// when params value is of type []string then last value is sent in request
 	resp, err := service.createAndSendRequest(ctx,
 		http.MethodPost, "test-path", map[string]interface{}{"key": "value", "name": []string{"gofr", "test"}},
-		[]byte("{Test Body}"), map[string]string{"header1": "value1"})
+		[]byte("{Test Body}"), map[string]string{"header1": "value1", "content-type": "application/json"})
```

Comment:
test needs to be updated with the following cases - 
1. We have added content-type but we are not verifying if the server which received the request received it with the content-type and the correct value.
2. We should be also adding a case where content-type is not set by the user and it automatically gets set.
3. Content-type is set with application/xml or something and it is not overrided.